### ISSUE TITLE
Call to a member function getConfiguration() on null

### ISSUE THREAD

**Issue by haruka** [phpstan/phpstan-doctrine#279]

**Comment by haruka** [comment]
Just for the completion: here a repo to reproduce the problem: https://github.com/haruka/phpstan_demo

**Comment by fcamp** [comment]
Any news about this? I got the same issue with 
```
phpstan/phpstan                     1.7.3   PHPStan - PHP Static Analysis Tool
phpstan/phpstan-doctrine            1.3.3    Doctrine extensions for PHPStan
```
while it goes fine with 
```
phpstan/phpstan                     1.7.2    PHPStan - PHP Static Analysis Tool
phpstan/phpstan-doctrine            1.3.3    Doctrine extensions for PHPStan
```

**Comment by ondrejmirtes** [comment]
You can work around this problem by configuring objectManagerLoader as said above.

**Comment by bestak** [comment]
For me it was reworking these annotations in the entity which was flagged by the phpstan:

Original with errors:
```php
#[ORM\Id]
#[ORM\Column(name: "connection_id", type: "integer", nullable: true)]
#[ORM\GeneratedValue(strategy: "SEQUENCE")]
#[ORM\SequenceGenerator(sequenceName: "connection__connection_id_seq", allocationSize:1, initialValue:1)]
private ?int $id = null;
```

Updated without phpstan errors:
```php
#[ORM\Id]
#[ORM\GeneratedValue]
#[ORM\SequenceGenerator(sequenceName: "connection__connection_id_seq")]
#[ORM\Column(name: "connection_id", type: "integer")]
private ?int $id = null;
```

**Comment by AlanCrevon** [comment]
Same problem here :

```
#[ORM\GeneratedValue(strategy: 'SEQUENCE')]
```

modified to 

```
#[ORM\GeneratedValue]
```

solved the problem

**Comment by orlandothoeny** [comment]
We ran into this issue when upgrading `doctrine/orm` [2.16.3](https://github.com/doctrine/orm/releases/tag/2.16.3) => [2.17.0](https://github.com/doctrine/orm/releases/tag/2.17.0).

The error occurred at https://github.com/doctrine/orm/pull/8931/files#diff-3e426fe47b58d2d4b970b1b7a0cbf5999af950e4f84e48dce4bee5061b57ed31R756

Changing the `GeneratedValue` strategy from `AUTO` to `IDENTITY` fixed the problem:
```
// From
#[ORM\GeneratedValue(strategy: 'AUTO')]
// To
#[ORM\GeneratedValue(strategy: 'IDENTITY')]
```
Note that we are using MySQL.
You might need to use either `SEQUENCE` or `IDENTITY`, depending on your database.
See https://github.com/greg0ire/doctrine-orm/blob/e89b680a285ea4c694cdaf9edb4490d904c23e54/UPGRADE.md#deprecated-reliance-on-the-non-optimal-defaults-that-come-with-the-auto-identifier-generation-strategy & https://github.com/greg0ire/doctrine-orm/blob/e89b680a285ea4c694cdaf9edb4490d904c23e54/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php#L76-L96

PHPStan output:
```
 -- --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
     Error                                                                                                                                                                                                                
 -- --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
     Internal error: Internal error: Call to a member function getConfiguration() on null while analysing file /var/www/admin/XXX/Comment.php                                            
                                                                                                                                                                                                                          
     Post the following stack trace to https://github.com/phpstan/phpstan/issues/new?template=Bug_report.yaml:                                                                                                            
     ## /var/www/admin/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php(742)                                                                                                                         
     #0 /var/www/admin/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php(625): Doctrine\ORM\Mapping\ClassMetadataFactory->determineIdGeneratorStrategy()                                              
     #1 /var/www/admin/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php(156): Doctrine\ORM\Mapping\ClassMetadataFactory->completeIdGeneratorMapping()                                                
     #2 /var/www/admin/vendor/doctrine/persistence/src/Persistence/Mapping/AbstractClassMetadataFactory.php(415): Doctrine\ORM\Mapping\ClassMetadataFactory->doLoadMetadata()                                             
     #3 /var/www/admin/vendor/doctrine/persistence/src/Persistence/Mapping/AbstractClassMetadataFactory.php(300): Doctrine\Persistence\Mapping\AbstractClassMetadataFactory->loadMetadata()                               
     #4 /var/www/admin/vendor/phpstan/phpstan-doctrine/src/Type/Doctrine/ObjectMetadataResolver.php(124): Doctrine\Persistence\Mapping\AbstractClassMetadataFactory->getMetadataFor()                                     
     #5 /var/www/admin/vendor/phpstan/phpstan-doctrine/src/Rules/Doctrine/ORM/PropertiesExtension.php(70): PHPStan\Type\Doctrine\ObjectMetadataResolver->getClassMetadata()                                               
     #6 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Node/ClassPropertiesNode.php(149): PHPStan\Rules\Doctrine\ORM\PropertiesExtension->isInitialized()                                                  
     #7 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Rules/Properties/MissingReadOnlyPropertyAssignRule.php(33): PHPStan\Node\ClassPropertiesNode->getUninitializedProperties()                          
     #8 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(107): PHPStan\Rules\Properties\MissingReadOnlyPropertyAssignRule->processNode()                                           
     #9 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(656): PHPStan\Analyser\FileAnalyser->PHPStan\Analyser\{closure}()                                                    
     #10 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(388): PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                         
     #11 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(627): PHPStan\Analyser\NodeScopeResolver->processStmtNodes()                                                        
     #12 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/NodeScopeResolver.php(357): PHPStan\Analyser\NodeScopeResolver->processStmtNode()                                                         
     #13 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Analyser/FileAnalyser.php(166): PHPStan\Analyser\NodeScopeResolver->processNodes()                                                                 
     #14 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(132): PHPStan\Analyser\FileAnalyser->analyseFile()                                                                       
     #15 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(97): PHPStan\Command\WorkerCommand->PHPStan\Command\{closure}()                         
     #16 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/clue/ndjson-react/src/Decoder.php(117): _PHPStan_c6b09fbdf\Evenement\EventEmitter->emit()                                                       
     #17 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(97): _PHPStan_c6b09fbdf\Clue\React\NDJson\Decoder->handleData()                         
     #18 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/Util.php(62): _PHPStan_c6b09fbdf\Evenement\EventEmitter->emit()                                                                
     #19 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/evenement/evenement/src/Evenement/EventEmitterTrait.php(97): _PHPStan_c6b09fbdf\React\Stream\Util::_PHPStan_c6b09fbdf\React\Stream\{closure}()  
     #20 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/react/stream/src/DuplexResourceStream.php(154): _PHPStan_c6b09fbdf\Evenement\EventEmitter->emit()                                               
     #21 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(201): _PHPStan_c6b09fbdf\React\Stream\DuplexResourceStream->handleData()                              
     #22 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/react/event-loop/src/StreamSelectLoop.php(173): _PHPStan_c6b09fbdf\React\EventLoop\StreamSelectLoop->waitForStreamActivity()                    
     #23 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/src/Command/WorkerCommand.php(98): _PHPStan_c6b09fbdf\React\EventLoop\StreamSelectLoop->run()                                                          
     #24 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Command/Command.php(259): PHPStan\Command\WorkerCommand->execute()                                                              
     #25 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(870): _PHPStan_c6b09fbdf\Symfony\Component\Console\Command\Command->run()                                       
     #26 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(261): _PHPStan_c6b09fbdf\Symfony\Component\Console\Application->doRunCommand()                                  
     #27 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/vendor/symfony/console/Application.php(157): _PHPStan_c6b09fbdf\Symfony\Component\Console\Application->doRun()                                         
     #28 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(124): _PHPStan_c6b09fbdf\Symfony\Component\Console\Application->run()                                                                      
     #29 phar:///var/www/admin/vendor/phpstan/phpstan/phpstan.phar/bin/phpstan(125): _PHPStan_c6b09fbdf\{closure}()                                                                                                       
     #30 /var/www/admin/vendor/phpstan/phpstan/phpstan(8): require('...')                                                                                                                                                 
     #31 /var/www/admin/bin/phpstan(117): include('...')                                                                                                                                                                  
     #32 {main}                                                                                                                                                                                                           
     Child process error (exit code 1):                                                                                                                                                                                   
 -- --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 


                                                                                                                        
 [ERROR] Found 2 errors                                                                                                 
                                                                                                                        
```
